Lib/webbrowser.py

```diff
@@ -588,14 +629,26 @@ def open(self, url, new=0, autoraise=True):
             return not rc
 
 
+def _get_supported_browsers():
+    _brow = []
+    for bt in _browsers_types:
+        try:
+            _ = get(bt)
+            _brow.append(bt)
+        except:
```

Comment:
Bare excepts should be avoided in 99% of cases, they can hide typos and other unintended things.  Catch the most specific error possible.

Reply:
i know

Reply:
have you tested how it works?